src/thread.cpp

```diff
@@ -158,10 +178,39 @@ void zmq::thread_t::setSchedulingParameters(int priority_, int schedulingPolicy_
 #endif
 
     posix_assert (rc);
+
+    if (use_nice_instead_priority &&
+            thread_priority != ZMQ_THREAD_PRIORITY_DFLT)
+    {
+        // assume the user wants to decrease the thread's nice value
+        // i.e., increase the chance of this thread being scheduled: try setting that to
+        // maximum priority.
+        rc = nice(-20);
+
+        // IMPORTANT: ignore return code: EPERM is typically returned for unprivileged processes
+        //            (CAP_SYS_NICE capability is required or RLIMIT_NICE resource limit should be set)
+        //            but in that case there is no much we can do
+
+    }
+
+    if (thread_affinity != ZMQ_THREAD_AFFINITY_DFLT)
+    {
+        cpu_set_t cpuset;
+        CPU_ZERO(&cpuset);
+        for (unsigned int cpuidx=0; cpuidx<sizeof(int)*8; cpuidx++)
+        {
+            int cpubit = (1 << cpuidx);
+            if ( (thread_affinity & cpubit) != 0 )
+                CPU_SET( cpuidx , &cpuset );
+        }
+        rc = sched_setaffinity(0 /* use calling thread PID */, sizeof(cpu_set_t), &cpuset);
+        posix_assert (rc);
+    }
 #else
 
     LIBZMQ_UNUSED (priority_);
```

Comment:
This function does not take parameters anymore, so this \#else branch and the 3 macros have to be removed

```
src/thread.cpp: In member function 'void zmq::thread_t::applySchedulingParameters()':
src/thread.cpp:211:20: error: 'priority_' was not declared in this scope
     LIBZMQ_UNUSED (priority_);
                    ^
src/macros.hpp:6:37: note: in definition of macro 'LIBZMQ_UNUSED'
 #define LIBZMQ_UNUSED(object) (void)object
                                     ^
src/thread.cpp:212:20: error: 'schedulingPolicy_' was not declared in this scope
     LIBZMQ_UNUSED (schedulingPolicy_);
                    ^
src/macros.hpp:6:37: note: in definition of macro 'LIBZMQ_UNUSED'
 #define LIBZMQ_UNUSED(object) (void)object
                                     ^
src/thread.cpp:213:20: error: 'affinity_' was not declared in this scope
     LIBZMQ_UNUSED (affinity_);
                    ^
src/macros.hpp:6:37: note: in definition of macro 'LIBZMQ_UNUSED'
 #define LIBZMQ_UNUSED(object) (void)object
```